README.md

```diff
@@ -52,33 +114,47 @@ task :ci => ['engine_cart:generate'] do
 end
 ```
 
-And in your e.g. spec\_helper.rb (or rails\_helper.rb), initialize EngineCart:
+And in your engine's test framework configuration (e.g. spec\_helper.rb or rails\_helper.rb), initialize EngineCart:
 
 ```ruby
   require 'engine_cart'
   EngineCart.load_application!
 ```
 
-When you want to clean out and regenerate the test application, you can call:
+Your test files (e.g. spec files for your engine) can now be written as tests for a Rails application.
+
+## Gotcha
+
+If you have generated a test application, there is a Gemfile and Gemfile.lock associated with the testing app at spec/internal (or wherever you designated as the test app location).  If you then update *your* engine's Gemfile or gemspec, Bundler can get confused if it has conflicting information between your engine and the testing app.
+
+To fix this:
+
+1. clean out your testing app `$ rm spec/internal`  # rake may not be avail.
```

Comment:
One case where rake is not available is when Bundler can't resolve the dependencies, because of conflicts between the test app Gemfile and the real Gemfile. Basically the situation @ndushay is trying to describes. 

That actually happens to me a lot, and is really frustrating, and I am never sure how to resolve it. So instructions for resolving it are welcome. 

And it's just that situation where rake can end up not available, because it won't run right without a `bundle exec` (as is common in Rails), but won't run with a `bundle exec` because bundler thinks it can't resolve the dependencies. 

So I think it's a real problem. 

(In general,  I understand basically why you have the main Gemfile including the test app Gemfile,  but have wondered sometimes if the cure is worse than the disease was -- it's the source of most of the challenges I have with engine_cart, and they can be very challenging). 

Thanks to @ndushay for trying to fill out the docs, I am optimistic that will help a lot with my use of engine_cart.